fdbrpc/SimExternalConnection.h

```diff
@@ -48,8 +48,19 @@ class SimExternalConnection final : public IConnection, public ReferenceCounted<
 	int write(SendBuffer const* buffer, int limit) override;
 	NetworkAddress getPeerAddress() const override;
 	UID getDebugID() const override;
+	static void addMockTCPEndpoint(const std::string& host,
+	                               const std::string& service,
+	                               const std::vector<NetworkAddress>& addresses);
+	static void updateMockTCPEndpoint(const std::string& host,
+	                                  const std::string& service,
+	                                  const std::vector<NetworkAddress>& addresses);
+	static void removeMockTCPEndpoint(const std::string& host, const std::string& service);
+	static void clearMockTCPEndpoints();
 	static Future<std::vector<NetworkAddress>> resolveTCPEndpoint(const std::string& host, const std::string& service);
 	static Future<Reference<IConnection>> connect(NetworkAddress toAddr);
+
+private:
+	static std::map<std::string, std::vector<NetworkAddress>> hostnameToAddresses;
```

Comment:
I feel this should be a global variable instead of a static member variable. Why doing it this way?

Reply:
If making it a global variable, then other classes, even the non-simulated ones, can also access it. I want it to only exist in simulations.